core/encoding.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2022 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package core
+
+import (
+	"fmt"
+
+	"github.com/sourcenetwork/defradb/client"
+	"github.com/sourcenetwork/immutable"
+)
+
+// Decode takes a field value and description and converts it to the
+// standardized Defra Go type.
+func Decode(fieldDesc client.FieldDescription, val any) (any, error) {
```

Comment:
suggestion: Now that this is the `core` package, it needs a more descriptive name, was tolerable when it was within `fetcher` lol.

Reply:
You don't think `Decode` with a FieldDescription is sufficient (possibly me still being too used overloading funcs)? Maybe `DecodeField`?

- [ ] Rename `Decode`

Reply:
Renamed to `DecodeFieldValue`